CeeFM/Decideify
# Suggestion Tags and Conditional Post Rednering Starting to Take Shape in Social.js

**Description**
You can now add posts to the database, and then also attach a suggestion to that post! The Post will render on the social.js page after reloading, and now if there is a suggestion attached to the post - the post will use the suggestion's image on the post rather than the post's image. This is just the beginning of how this will work but happy to have cleared the litany of errors that were preventing it from even rendering earlier today!

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On social.js you can:
  - Fill out the post form, submit it, and it adds to the database
  - You can pick a suggestion to add to that post from a modal that pops up thereafter, or you can continue. Either way the page will reload.
  - Your post will render as the top post on that page after the reload. 
  - If a suggestion is tagged to a post, the post will show the imageLocation of the suggestion rather than the post. This may or may not stay as such, but is the first step in getting conditional rendering working for posts with suggestions tagged to them. More updates soon! 
  - Comments haven't even been created or added to database yet though.

Checklist
